### PR TITLE
Add GetReplicas API for supporting memory replica

### LINKED ISSUE #272: [Feature]: Support MultipleReplica in milvus go sdk
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Milvus 2.1 allows users to load collection/partitions as multiple replicas to utilize extra querynodes' CPU and memory. With this ability, the overall qps and throughput can be scaled out with extra hardware.

Go SDK should support this feature. 

### Describe the solution you'd like.

* [x] Add ReplicaNumber option to specify replica number in collection loading #273
* [x] Add GetReplicas API to check information about replica groups, as well as their corresponding querynodes and shards #274

### Describe an alternate solution.

_No response_

### Anything else? (Additional Context)

_No response_

### PR DESCRIPTION
**Pull request by letian-jiang**
Add GetReplicas API for supporting memory replica
Close #272 

Signed-off-by: Letian Jiang <letian-jiang@zilliz.com>

### CODE REVIEW

File: client/client_grpc_collection.go [hunk 1]
```diff
@@ -394,3 +395,46 @@ func (c *grpcClient) ReleaseCollection(ctx context.Context, collName string) err
...
+	}
+
+	req := &server.GetReplicasRequest{
+		CollectionID:   coll.ID,
+		WithShardNodes: true, // return nodes by default
+	}
+
+	resp, err := c.service.GetReplicas(ctx, req)
+	if err != nil {
+		return nil, err
```
Comment (congqixia) [inline client/client_grpc_collection.go:416]: Please add unit test to cover grpc error case.

File: client/client_grpc_collection.go [hunk 2]
```diff
@@ -394,3 +395,46 @@ func (c *grpcClient) ReleaseCollection(ctx context.Context, collName string) err
...
+	}
+	if err = handleRespStatus(resp.GetStatus()); err != nil {
+		return nil, err
```
Comment (congqixia) [inline client/client_grpc_collection.go:419]: Please add unit test to cover server return error status.

File: client/client_grpc_collection.go (hunk 1 above)
Comment (letian-jiang) [reply client/client_grpc_collection.go:416]: Done

File: client/client_grpc_collection.go (hunk 2 above)
Comment (letian-jiang) [reply client/client_grpc_collection.go:419]: Done

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**